ETM_Sketch/Tile.pde

```diff
@@ -0,0 +1,130 @@
+class Tile {
+  private final static int overlapTolerance = 25;
+  Pointer pointer;
+  
+  float r, rx, ry;
+  
+  private int offsetX;
+  private int offsetY;
+  
+  PGraphics pg;
+  PImage tex;
+  PShader deform;
+  private int posX;
+  private int posY;
+  private int sizeX;
+  private int sizeY;
+  
+  float alpha;
+  //int o, g;
+  
+  Tile (PImage t, Pointer p, int i, int j, int x, int y, int w, int h) {
+    this.pointer = p;
+    this.posX = x;
+    this.posY = y;
+    this.sizeX = w;
+    this.sizeY = h;
+    this.tex = t;
+    this.deform = loadShader("deform.glsl", "vert.glsl");
+    this.pg = createGraphics(this.sizeX, this.sizeY, P2D);
+    this.deform.set("resolution", float(width), float(height));
+    float offsetX = i * 2;
+    float offsetY = j * -2;
+    this.deform.set("offset", offsetX, offsetY, 0f, 0f);
+    
+    rx = this.posX + w/2;
+    ry = this.posY + h/2;
+    r = dist(this.posX, this.posY, rx, ry);
+  }
+  
+  boolean pointerOverlaps(){
+    float d = dist(rx, ry, this.pointer.x, this.pointer.y);
+    if (d < r + this.pointer.radius) {
+     return true;
+    } else {
+      return false;
+    }
+  }
+  
+  void update() {
+    if (this.pointer.x > this.posX && this.pointer.x < this.posX + this.sizeX/2) {
+      this.alpha = int(map(this.pointer.x, this.posX, this.posX + this.sizeX/2, 25, 100));
+      //o = 255;
+      //g = 0;
+    } 
+    if (this.pointer.x > this.posX + this.sizeX/2 && this.pointer.x < this.posX + this.sizeX) {
+      this.alpha = int(map(this.pointer.x, this.posX + this.sizeX/2, this.posX + this.sizeX, 100, 25));
+      //o = 0;
+      //g = 255;
```

Comment:
Das ist auch komplett von dir übernommen die logik für den `alpha` wert

Reply:
Die Verwendung des Alpha wertes war ja nur ein Hilfsmittel für mich um zu testen ob ich die `pointerOverlaps` in eine "Reaktion" übersetzten kann. D.h. du hast this.alpha ( jetzt dann umgeschrieben damit es nicht mehr die opacity steuert sondern eben `w` im shader? Oder ist das eh nur relevant für `SHOW_OVERLAP_GRID = true;`?